specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2220 @@
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/BuildDefinition"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/BuildDefinition"
+            }
+          },
+          "404": {
+            "description": "Not Found"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "BuildDefinitions_Update": {
+            "$ref": "./examples/BuildDefinitionsUpdate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/buildDefinitions/{buildDefinitionName}/queue": {
+      "post": {
+        "tags": [
+          "BuildDefinitions"
+        ],
+        "summary": "Creates a new build against the given build definition and add it to the build queue.",
+        "operationId": "BuildDefinitions_QueueBuild",
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/registryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/buildDefinitionNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Build"
+            }
+          },
+          "202": {
+            "description": "Success"
+          },
+          "404": {
+            "description": "Not Found"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "BuildDefinitions_QueueBuild": {
+            "$ref": "./examples/BuildDefinitionsQueueBuild.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/buildDefinitions/{buildDefinitionName}/addTrigger": {
+      "post": {
+        "tags": [
+          "BuildDefinitions"
+        ],
+        "summary": "Adds a build trigger to a build definition.",
+        "operationId": "BuildDefinitions_AddTrigger",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "buildTriggerParameters",
+            "in": "body",
+            "description": "The parameters for adding a build trigger.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BuildTriggerParameters"
```

Comment:
>BuildTriggerParameters [](start = 37, length = 22)

Looks like it is equivalent to `BuildTrigger`, why not get rid of `BuildTriggerParameters`?

Reply:
The trigger parameters doesn't have TriggerId that will given by the RP. BuildTrigger is the response for GET and has the trigger id to operate upon the trigger.